### DISCUSSION TITLE
How do I add requiresConfirmation in ToggleColumn and SelectColumn in Table?

### DISCUSSION THREAD

**Discussion by yuvrajtimalsina** [filamentphp/filament#8693] [General]
Is there a way to add requiresConfirmation modal in these columns?

```
Tables\Columns\ToggleColumn::make('status'),
Tables\Columns\SelectColumn::make('category_id')
```

**Comment by pboivin** [comment]
I don't think this is possible. Maybe you could use a column action instead, to show a modal:
https://filamentphp.com/docs/3.x/tables/columns/getting-started#action-modals

**Reply by yuvrajtimalsina** [reply]
Clicking toggles the switch, means I have to prevent its toggling behaviour and toggle it using modal action instead.

🤔 Seems pretty Scary 😅

**Reply by pboivin** [reply]
I'm suggesting to replace the toggle entirely with a modal. In other words, you would use a TextColumn in the table with a modal action on it. In the modal, you can display your toggle. From there, the user can click Save or Cancel.

**Reply by yuvrajtimalsina** [reply]
Hope, it will be added as a feature in near future. I may now proceed with the available options.

**Reply by myregistration** [reply]
I'd like ability to add `requiresConfirmation()` to `SelectColumn` as well

**Reply by yuvrajtimalsina** [reply]
> I'd like ability to add `requiresConfirmation()` to `SelectColumn` as well

It would be better I guess.
